conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptions.java

```diff
@@ -199,6 +198,34 @@ private static void deadlineExpiredException(HttpServerExchange exchange, Deadli
                 exception, exchange, UndertowDeadlineReasonResponseEncodingAdapter.INSTANCE);
     }
 
+    private static void ioException(HttpServerExchange exchange, IOException ioException) {
+        if (!exchange.getConnection().isOpen()) {
+            log.info(
+                    "I/O exception from a closed connection. The request may have been aborted by the client",
+                    ioException);
+            return;
+        }
+
+        if (ioException
+                .getMessage()
+                .equals(UndertowMessages.MESSAGES
+                        .couldNotReadContentLengthData()
+                        .getMessage())) {
```

Comment:
`ioException.getMessage()` could return `null` leading to a `NullPointerException` on this error handling path, so we need to handle that.

Additionally, `UndertowMessages.MESSAGES.couldNotReadContentLengthData()` creates a new `IOException` on each invocation and adds the expensive overhead of `Throwable#fillInStackTrace()`. We should resolve the message once to a static final field.

I've pushed https://github.com/palantir/conjure-java/pull/2523 as suggested changes on top of this branch

```suggestion
        if (Objects.equals(ioException.getMessage(), COULD_NOT_READ_CONTENT_LENGTH_MESSAGE)) {
```